### PR TITLE
Fix typescript-node generation of array type models

### PR DESCRIPTION
**Pull request by bvwells**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny

### Description of the PR

Fix for generating models of type array. If the model is of type array do not generate getAttributeTypeMap function and attributeTypeMap. Should address issue #7827.

### CODE REVIEW

File: samples/client/petstore/typescript-node/default/.swagger-codegen/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.3.1
\ No newline at end of file
+unset
```
Comment (macjohnny) [inline samples/client/petstore/typescript-node/default/.swagger-codegen/VERSION]: should be 2.4.0-SNAPSHOT

Comment (bvwells) [reply samples/client/petstore/typescript-node/default/.swagger-codegen/VERSION]: This was quite strange for me. Whenever I have regenerated clients/servers in the past this version has been automatically set when running the generation scripts in /bin. I'm wondering whether something has changed/broken. Any ideas?

**Comment by bvwells** [comment]
@macjohnny thanks for the review.

I tested the change by adding a couple of models to [petstore.yaml](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml) and it seemed to be working fine.

I'd quite like to add a test so that this functional doesn't break. Any suggestions on this?

**Comment by bvwells** [comment]
I had to update the swagger version manually. Not sure why this isn't being automatically updated when I run the scripts in \bin. Will investigate this separately.

**Pull request closed without merging**